packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -0,0 +1,283 @@
+		 * @param {Object} settings The current settings.
+		 * @public
+		 */
+		"cancel-click": {},
+	},
+};
+
+/**
+ * @class
+ * <h3 class="comment-api-title">Overview</h3>
+ * The <code>ui5-view-settings-dialog</code> component consisting of several lists.
+ * One of them (<code>Sort order</code>) is built-in, and another (<code>Sort By</code>) must be provided by the developer.
+ * The selected options can be used to create sorters for the table.
+ *
+ * The <code>ui5-view-settings-dialog</code> interrupts the current app processing as it is the only focused UI element and
+ * the main screen is dimmed/blocked.
+ * The <code>ui5-view-settings-dialog</code> is modal, which means that user action is required before returning to the parent window is possible.
+ *
+ * <h3>Structure</h3>
+ * A <code>ui5-view-settings-dialog</code> consists of a header, content, and a footer for action buttons.
+ * The <code>ui5-view-settings-dialog</code> is usually displayed at the center of the screen.
+ *
+ * <h3>Responsive Behavior</h3>
+ * <code>ui5-view-settings-dialog</code> stretches on full screen on phones.
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents/dist/ViewSettingsDialog";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.fiori.ViewSettigsDialog
+ * @extends UI5Element
+ * @tagname ui5-view-settings-dialog
+ * @public
+ */
+class ViewSettingsDialog extends UI5Element {
+	constructor() {
+		super();
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents-fiori");
+		this._initialSettings = {}; // settings when the control is opened for the first time
+		this._currentSettings = {}; // settings saved after pressing OK button
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get dependencies() {
+		return [
+			Bar,
+			Button,
+			Dialog,
+		];
+	}
+
+	static get template() {
+		return ViewSettingsDialogTemplate;
+	}
+
+	static get styles() {
+		return viewSettingsDialogCSS;
+	}
+
+	static async onDefine() {
+		await fetchI18nBundle("@ui5/webcomponents-fiori");
+	}
+
+	get _dialogTitle() {
+		return this.i18nBundle.getText(VSD_DIALOG_TITLE_SORT);
+	}
+
+	get _okButtonLabel() {
+		return this.i18nBundle.getText(VSD_SUBMIT_BUTTON);
+	}
+
+	get _cancelButtonLabel() {
+		return this.i18nBundle.getText(VSD_CANCEL_BUTTON);
+	}
+
+	get _resetButtonLabel() {
+		return this.i18nBundle.getText(VSD_RESET_BUTTON);
+	}
+
+	get _ascendingLabel() {
+		return this.i18nBundle.getText(VSD_ORDER_ASCENDING);
+	}
+
+	get _descendingLabel() {
+		return this.i18nBundle.getText(VSD_ORDER_DESCENDING);
+	}
+
+	get _sortOrderLabel() {
+		return this.i18nBundle.getText(VSD_SORT_ORDER);
+	}
+
+	get _sortByLabel() {
+		return this.i18nBundle.getText(VSD_SORT_BY);
+	}
+
+	get _isPhone() {
+		return isPhone();
+	}
+
+	get _sortAscending() {
+		return !this.sortDescending;
+	}
+
+	get _sortDescending() {
```

Comment:
You can just use the variable in the template. This getter is not needed

Reply:
done